src/hardware/pounder/mod.rs

```diff
@@ -383,40 +428,42 @@ impl PounderDevices {
     pub fn set_gpio_pin(
         &mut self,
         pin: GpioPin,
-        state: bool,
+        level: mcp230xx::Level,
     ) -> Result<(), Error> {
         self.mcp23017
-            .digital_write(pin as _, state)
-            .map_err(|_| Error::I2c)?;
-        Ok(())
+            .set_gpio(pin.into(), level)
+            .map_err(|_| Error::I2c)
     }
 
     /// Select external reference clock input.
     pub fn set_ext_clk(&mut self, enabled: bool) -> Result<(), Error> {
+        let level = if enabled {
+            mcp230xx::Level::High
+        } else {
+            mcp230xx::Level::Low
+        };
         // Active low
-        self.set_gpio_pin(GpioPin::OscEnN, enabled)?;
-        self.set_gpio_pin(GpioPin::ExtClkSel, enabled)
+        self.set_gpio_pin(GpioPin::OscEnN, level)?;
+        self.set_gpio_pin(GpioPin::ExtClkSel, level)
     }
 }
 
 impl attenuators::AttenuatorInterface for PounderDevices {
     /// Reset all of the attenuators to a power-on default state.
     fn reset_attenuators(&mut self) -> Result<(), Error> {
         // Active low
-        self.set_gpio_pin(GpioPin::AttRstN, false)?;
-        self.set_gpio_pin(GpioPin::AttRstN, true)
+        self.set_gpio_pin(GpioPin::AttRstN, mcp230xx::Level::Low)?;
+        self.set_gpio_pin(GpioPin::AttRstN, mcp230xx::Level::High)
     }
 
     /// Latch a configuration into a digital attenuator.
     ///
     /// Args:
     /// * `channel` - The attenuator channel to latch.
     fn latch_attenuator(&mut self, channel: Channel) -> Result<(), Error> {
-        let pin =
-            GpioPin::try_from(GpioPin::AttLe0 as u8 + channel as u8).unwrap();
-        // Active low
-        self.set_gpio_pin(pin, false)?;
-        self.set_gpio_pin(pin, true)
+        // Rising edge sensitive
+        self.set_gpio_pin(channel.into(), mcp230xx::Level::High)?;
```

Comment:
This looks like it inverted the low -> high logic that we had previously to generate a rising edge. This wouldn't generate an edge if the IO were already high.

Reply:
yes. It's inverted because that's what the datasheet uses.
The second comment also applied previously. But it's not a problem since the state is know to be correct.

Reply:
Indeed, it appears that the device is infact rising edge sensitive.

While this code may be correct now, I feel that it would be easy to break in the future (e.g. by initializing IOs high instead of low). We can mitigate that and _guarantee_ that a rising edge is generated regardless of initial pin state by first setting low, then high.

Reply:
Right. That's also why I changed it to this sequence a while ago.
https://github.com/quartiq/stabilizer/commit/fc78b087eeba5343d25dd90f729e4bf1a5180dd9
I'll try again with that robust sequence and see.

The reverted sequence here is the result of trying to hunt down the mcp23017 issues (which I still haven't fully done)...